app/repositories/SchemeCacheRepository.scala

```diff
@@ -0,0 +1,210 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package repositories
+
+import java.nio.charset.StandardCharsets
+
+import com.google.inject.Inject
+import org.joda.time.{DateTime, DateTimeZone}
+import play.api.libs.json.{Format, JsValue, Json, _}
+import play.api.{Configuration, Logger}
+import play.modules.reactivemongo.ReactiveMongoComponent
+import reactivemongo.api.indexes.{Index, IndexType}
+import reactivemongo.bson.Subtype.GenericBinarySubtype
+import reactivemongo.bson.{BSONBinary, BSONDocument, BSONObjectID}
+import reactivemongo.play.json.ImplicitBSONHandlers._
+import uk.gov.hmrc.crypto.{Crypted, CryptoWithKeysFromConfig, PlainText}
+import uk.gov.hmrc.mongo.ReactiveRepository
+import uk.gov.hmrc.mongo.json.ReactiveMongoFormats
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class SchemeCacheRepository @Inject()(collectionName: String,
```

Comment:
Is there any way the code shared between this and PensionSchemeCacheRepository could be put in one place and shared rather than duplicated?